cmake/Macros.cmake

```diff
@@ -274,6 +274,10 @@ macro(sfml_add_library module)
         target_compile_definitions(${target} PUBLIC "SFML_STATIC")
     endif()
 
+    # Enable support for UTF-8 characters in source code
+    if(SFML_COMPILER_MSVC)
+        target_compile_options(${target} PRIVATE /utf-8)
```

Comment:
This is needed only for tests, right? Why adding it for libraries and examples too?
Also, the comment could tell why this option is needed.

Reply:
> This is needed only for tests, right?

Correct. It's because of the viralitiy of pre-compiled headers. For those to work all TUs must use the same arguments. Not my preference but otherwise CI would fail.

Reply:
I guess one question is whether `/source-charset:utf-8` is enough. Are they an downsides to using utf8 in the executing charset?

https://learn.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8?view=msvc-170

Reply:
https://github.com/ChrisThrasher/SFML/actions/runs/13162292727/job/36733841275

CI is upset if we do that.